src/reducers/comments.spec.tsx

```diff
@@ -19,35 +25,80 @@ describe(__filename, () => {
 
-      const state = reducer(
-        undefined,
-        actions.beginComment({ versionId, fileName, line }),
+      const state = reducer(undefined, actions.beginComment(params));
+
+      expect(state.byKey[createCommentKey(params)]).toEqual(
+        createCommentInfo({ beginNewComment: true }),
       );
+    });
 
-      expect(
-        state.byKey[createCommentKey({ versionId, fileName, line })],
-      ).toEqual(createCommentInfo({ beginNewComment: true }));
+    it('aborts saving a comment by key', () => {
+      const params = { versionId: 1, fileName: 'manifest.json', line: 123 };
+
+      let state;
+
+      // Imagine that this happened some time in the past.
+      state = reducer(state, actions.beginSaveComment(params));
+      // Imagine that the user opened a comment form again.
+      state = reducer(state, actions.beginComment(params));
+
+      expect(state.byKey[createCommentKey(params)]).toMatchObject({
+        savingComment: false,
+      });
+    });
+  });
+
+  describe('beginSaveComment', () => {
+    it('begins saving a comment by key', () => {
+      const params = { versionId: 1, fileName: 'manifest.json', line: 123 };
+
+      const state = reducer(undefined, actions.beginSaveComment(params));
+
+      expect(state.byKey[createCommentKey(params)]).toEqual(
+        createCommentInfo({ savingComment: true }),
+      );
+    });
+  });
+
+  describe('abortSaveComment', () => {
+    it('aborts saving a comment by key', () => {
+      const params = { versionId: 1, fileName: 'manifest.json', line: 123 };
+
+      let state;
+
+      state = reducer(state, actions.beginSaveComment(params));
+      state = reducer(state, actions.abortSaveComment(params));
+
+      expect(state.byKey[createCommentKey(params)]).toEqual(
+        createCommentInfo({ savingComment: false }),
+      );
     });
   });
 
   describe('finishComment', () => {
     it('finishes a comment by key', () => {
-      const fileName = 'manifest.json';
-      const line = 123;
-      const versionId = 1;
+      const params = { versionId: 1, fileName: 'manifest.json', line: 123 };
 
-      const state = reducer(
-        undefined,
-        actions.finishComment({ fileName, line, versionId }),
+      const state = reducer(undefined, actions.finishComment(params));
+
+      expect(state.byKey[createCommentKey(params)]).toEqual(
+        createCommentInfo({ beginNewComment: false }),
       );
+    });
 
-      expect(
-        state.byKey[createCommentKey({ versionId, fileName, line })],
-      ).toEqual(createCommentInfo({ beginNewComment: false }));
+    it('aborts saving a comment', () => {
+      const params = { versionId: 1, fileName: 'manifest.json', line: 123 };
+
+      let state;
+
+      state = reducer(state, actions.beginSaveComment(params));
+      state = reducer(state, actions.finishComment(params));
+
+      expect(state.byKey[createCommentKey(params)]).toMatchObject({
```

Comment:
Oh, I see you do use `toMatchObject` here, rather than `toEqual(createCommentInfo())`, which is similar to what I asked about regarding `expect.objectContaining()`. Why did you decide to do that here?

Reply:
The first test you commented on only calls one action so it's safe to use `createCommentInfo()`, the common initializer for the state. I used `expect.objectContaining()` here, instead, because multiple actions are involved; I don't want to couple the test to unrelated state changes triggered by those other actions now or in the future.